BD-3167: Track user sync beta - MERGE 6/18 (#7515)
braze-inc/braze-docs: _docs/_hidden/private_betas/post_user_track_synchronous.md

## Request body

```
Content-Type: application/json
Authorization: Bearer YOUR_REST_API_KEY
```

```json
{
  "attributes": (optional, one attributes object),
  "events": (optional, one event object),
  "purchases": (optional, one purchase object),
}
```

### Request parameters

{% alert important %}
For each request component listed in the following table, one of `external_id`, `user_alias`, `braze_id`, `email`, or `phone` is required.
{% endalert %}

| Parameter | Required | Data Type | Description |
| --------- | ---------| --------- | ----------- |
| `attributes` | Optional | One attributes object | See [user attributes object]({{site.baseurl}}/api/objects_filters/user_attributes_object/) |
| `events` | Optional | One event object | See [events object]({{site.baseurl}}/api/objects_filters/event_object/) |
| `purchases` | Optional | One purchase object | See [purchases object]({{site.baseurl}}/api/objects_filters/purchase_object/) |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3  .reset-td-br-4}

## Responses

> When using any of the aforementioned API requests, you should receive one of the following two general responses: a successful message or a message with fatal errors.

Reviewer: ```suggestion
When using this endpoint's [request parameters](#request-parameters), you should receive one of the following responses: a successful message or a message with fatal errors.
```

Reviewer: ^ the goal being to replace the word `aforementioned`.

### Successful message

> Successful messages will be met with the following response, which includes information about the user profile data that was updated.

Reviewer: ```suggestion
Successful messages will return the following response, which includes information about the user profile data that was updated.
```

          "alias_name" : "device123", 
          "alias_label" : "my_device_identifier"
      }
      "app_id" : "11ae5b4b-2445-4440-a04f-bf537764c9ad",
      "product_id" : "Completed Order",
      "currency" : "USD",
      "price" : 219.98,
      "time" : "2022-12-06T19:20:45+01:00",
      "properties" : {
          "products" : [ 
            {
              "name": "Monitor",
              "category": "Gaming",
              "product_amount": 19.99
            },
            { 
              "name": "Gaming Keyboard",
              "category": "Gaming ",
              "product_amount": 199.99
            }
          ]
      }
   }
  ]
}'
```

#### Response

```
{
    "users": [
        {
          "user_alias" : { 
            "alias_name" : "device123", 
            "alias_label" : "my_device_identifier"
          },
          "purchase_events": [
                {
                "product_id": "Completed Order",
                "first": "2013-07-16T19:20:30+01:00",
                "last": "2022-12-06T18:20:45.000Z",
                "count": 3
                }
            ]
        }
    ],
    "message": "success"
} 
```

## Frequently asked questions

> ### In what instances should I use the asynchronous or synchronous endpoint?

Reviewer: ```suggestion
### Should I use the asynchronous or synchronous endpoint?
```

For most profile updates, the `/users/track` endpoint will work best because of its higher rate limit and flexibility to let you batch requests. However, the `/users/track/sync` endpoint is useful if you're experiencing race conditions due to rapid, consecutive requests for the same user.

> ### Does response time differ compared to the `/users/track` endpoint?

Reviewer: ```suggestion
### Does response time differ from the `/users/track` endpoint?
```

With a synchronous call, the API waits until the request is completed to return a response. As a result, synchronous requests will take longer on average than asynchronous requests to `/users/track`. For the majority of requests, you can expect a response within seconds.

> ### Can you send multiple requests to this endpoint at the same time?

Reviewer: ```suggestion
### Can I send multiple requests to this endpoint at the same time?
```

This endpoint has a lower rate limit than the `/users/track` endpoint (see [rate limit](#rate-limit) below). Each `/users/track/sync` request can contain only  one event object, one attribute object, **or** one purchase object. This endpoint should be reserved for user profile updates where a synchronous call is needed. For a healthy implementation, we recommend using `/users/track/sync` and `/users/track` together.

For example, if you're sending consecutive requests for the same user over a short period of time, race conditions are possible with the asynchronous `/users/track` endpoint, but with the `/users/track/sync` endpoint you can send those requests in sequence, each after receiving a `2XX` response.

## Prerequisites

To use this endpoint, you'll need an [API key]({{site.baseurl}}/api/api_key/) with the `users.track.sync` permission.

Customers using the API for server-to-server calls may need to allowlist `rest.iad-01.braze.com` if they're behind a firewall.

## Rate limit

> We apply a base speed limit of 500 requests per minute to this endpoint for all customers. Each `/users/sync/track/` request can contain up to one event object, one attribute object, or one purchase object. Each object (event, attribute, and purchase arrays) can update one user each.

Reviewer 2: Should be /users/track/sync

{% alert note %}

> Reach out to your customer success manager if you need your limit increased.

Reviewer 2: We should just take this line out, while I definitely want to hear feedback on whether this rate limit is too small, we can't guarantee increasing it.
